patroni/ha.py

```diff
@@ -1099,7 +1126,10 @@ def handle_long_action_in_progress(self):
                             self.state_handler.terminate_starting_postmaster(postmaster=task.result)
                 self.demote('immediate-nolock')
                 return 'lost leader lock during ' + self._async_executor.scheduled_action
-
+            if self._async_executor.scheduled_action == 'promote' and self.state_handler.pre_promote_script:
```

Comment:
Since the `pre_promote_script` is a property, it's value might change in runtime...
This is especially bad when it was returning something at the moment when the script was started and later it become empty.